### ISSUE TITLE
Finemap

### ISSUE THREAD

**Issue by jameskozubek** [richardslab/Ei#2]
Thanks for the code. 
I am stuck at this step 03_run-finemap.sh
I guess I should download this?
http://www.christianbenner.com/

**Comment by richardslab** [comment]
Correct. You will also need LDStore, as well as bgenix, which is part of bgen, found here https://enkre.net/cgi-bin/code/bgen/dir?ci=trunk.

Also, I just realized the Makefile is in the wrong directory (02_pair-genes-to-snv). It should be in same directory as 03_run-finemap.sh. I will fix this and push edit.

I very much appreciate your interest and realize the code requires some refactoring, which I continue to work on and would very much appreciate your feedback.

**Comment by jameskozubek** [comment]
I ran LD clumping using a random panel and my GWAS stats. Is the idea that I run LDStore only once for all a) only lead snps, or b) for all 500K snps? or c) once for each chromosomes? to create my *ld file, so should i have many *ld files, or only one, and is it only for the lead snps?
and, then, am I supposed to run Finemap, for each chromosome, for each run, should the GWAS file be limited to just the Lead snps, or all 500K snps per each chromosome?

**Comment by richardslab** [comment]
This largely depends on resources available on your end. I chose to:

1. Determine loci from lead SNPs (see Makefile) as roughly 1 MB around each lead SNP, but merging adjacent lead SNPs prior to padding.
2. I then run LDStore on each of these loci.
3. I then run FINEMAP for each locus.

Doing it this way I was able to parallelize the process onto a HPC using a batch scheduler. 

Alternatively, as you suggest, you can run LDStore on the entire genome and keep that as a single data set from which you draw SNP correlations.

By 500k SNPs, I assume you mean 500k UKB individuals. If so, I used a 50k random subset from how Bycroft et al defined the white British subset. This flag is available in the ukb_sqc_v2.txt file from UK Biobank (column in.white.ancestry.subset). I can share code with you how I did this, but was simply some `awk` followed by `sort -R`. Main reason for 50k is computational, and also using all 500k would likely yield marginal improvement.

**Comment by jameskozubek** [comment]
I see. that makes sense.

**Comment by jameskozubek** [comment]
I tried it two ways, once running LDStore per each locus + 1MB, and then again running it by combining all of these SNPs into a single file (15K snps in total in single LD store file). However, if I do it as a single file, Finemap creates a resulting "cred" file in which there are SNPs from multiple different chromosomes in the same causal set (say a set of 5 snps but one is on chr21 and another on chr1). Can this be meaningful, or is it nonsense to use Finemap in this way?

**Comment by richardslab** [comment]
15k SNPs in total across entire genome? If so, where the SNPs filtered a priori? Normally SNPs should not be agressively filtered prior to FINEMAP. For instance, each of the 1Mbp loci we processsed for Ei contained ~1000 SNPs. It would be good to know more specifics on your analysis if you can provide. If you feel uncomfortable doing so publicly you can send me an email.

**Comment by jameskozubek** [comment]
I thought the idea was you select your top hits gwas SNPs, so say 35 snps. then you go out 1MB. Then I have 35 files created from LDStore each with about ~500 to 1000 snps. then I run finemap on each of the 35 files separately. I also tried to run finemap by combining all 35 loci + 1MB into a single LD file, which is about 15K snps.

**Comment by jameskozubek** [comment]
First: I am wondering if this makes sense, to create a single LD file which incorporates snps from different loci on different chromosomes, or if it only is appropriate to create separate LD files for each loci. Second question: How do I evaluate what is a good posterior probability? Should I only trust the cred files with > 0.95 as it says in the log file? Or, can I trust all the results in the cred files. Thank you for your time!

**Comment by jameskozubek** [comment]
james@precisionlife.com

**Comment by richardslab** [comment]
Got it. Apologies for my confusion. I kept everything separate. Unclear from the FINEMAP documentation whether you can merge them prior to analysis.

**Comment by jameskozubek** [comment]
I agree. It's an interesting question. I will see if the Finemap team has any opinions on this.

**Comment by richardslab** [comment]
> 
> 
> First: I am wondering if this makes sense, to create a single LD file which incorporates snps from different loci on different chromosomes, or if it only is appropriate to create separate LD files for each loci. Second question: How do I evaluate what is a good posterior probability? Should I only trust the cred files with > 0.95 as it says in the log file? Or, can I trust all the results in the cred files. Thank you for your time!

I would *potentially* make a single LDStore file simply to have a reference dataset from which you can pull the correlations from, but I would run FINEMAP per locus.

Filtering FINEMAP results is subjective to some extent. log10(BF)>2 is typically used and has shown good enrichment for functional genomic annotations and proximity to causal genes. Likely same for cred > 0.95. Of note, we removed loci that failed to converge, ie where the posterior probability of causal config containing k snps is > 0.95 at max_k.
